src/dos/drive_local.cpp

```diff
@@ -457,10 +457,10 @@ bool localDrive::GetFileAttr(char* name, FatAttributeFlags* attr)
 	CROSS_FILENAME(newname);
 	dirCache.ExpandNameAndNormaliseCase(newname);
 
-	const auto result = local_drive_get_attributes(newname, *attr);
-	if (result != DOSERR_NONE) {
+	if (local_drive_get_attributes(newname, *attr) != DOSERR_NONE) {
+		// The caller is responsible to act accordingly, possibly
+		// it should set DOS error code (setting it here is not allowed)
 		*attr = 0;
-		DOS_SetError(result);
 		return false;
 	}
```

Comment:
Also confirmed that the other derived drives `GetFileAttr()`'s also aren't setting `DOS_Error`:
---
src/dos/drive_fat.cpp:bool fatDrive::GetFileAttr(char* name, FatAttributeFlags* attr)
src/dos/drive_iso.cpp:bool isoDrive::GetFileAttr(char* name, FatAttributeFlags* attr)
src/dos/drive_local.cpp:bool localDrive::GetFileAttr(char* name, FatAttributeFlags* attr)
src/dos/drive_local.cpp:bool cdromDrive::GetFileAttr(char* name, FatAttributeFlags* attr)
src/dos/drive_overlay.cpp:bool Overlay_Drive::GetFileAttr(char* name, FatAttributeFlags* attr)
src/dos/drive_virtual.cpp:bool Virtual_Drive::GetFileAttr(char* name, FatAttributeFlags* attr)
---

Found with: `git grep '^bool .*::GetFileAttr'`

Reply:
Yes - this was a copy&paste development mistake.